pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -389,14 +391,29 @@ func (d *DeploymentController) render(templateName string, mi TemplateInput) ([]
 	if template == nil {
 		return nil, fmt.Errorf("no %q template defined", templateName)
 	}
+	proxyConfig := cfg.MeshConfig.GetDefaultConfig()
+	if d.env.PushContext != nil && d.env.PushContext.ProxyConfigs != nil {
+		if generatedProxyConfig := d.env.PushContext.ProxyConfigs.EffectiveProxyConfig(
```

Comment:
certainly not cause by this PR, just a note:  EffectiveProxyConfig taking NodeMetadata is not great. Should jsut take a struct with namespace,labels,annotations. Otherwise its hard to tell what you do or don't need.

Reply:
Done - I encapsulated this logic to a dedicated function.